### PR TITLE
feat(OGC-1142): add domain column to panel schema (M1)

### PR DESCRIPTION
**Pull request by Agaba-derrick**
# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the
      Issue number if applicable.
- [ ] The PR includes a video showing the changes for the work done.
- [ ] The PR title follows conventional commit label standards.
- [ ] The changes confirm to the OpenElis Global x3
      [Styleguide and Design](https://uwdigi.atlassian.net/wiki/spaces/OG/pages/621346838/OpenELIS+Global+Style+Guide)
      documentation.
- [ ] The changes include tests or are validated by existing tests.
- [ ] I have read and agree to the Contributing
      [Guidelines](https://openelis-global.org/community/get-involved/) of this
      project.

## Summary

This PR implements **Milestone 1 (Schema & Backend Foundation)** for the Panel Management Domain Upgrade (OGC-1142) aligning with **FRS v2.2**. 

Following the Director's architectural feedback, this PR abandons the v1 schema (which linked panels to lab units and required unique codes) in favor of the new domain-scoped architecture where LOINC serves as the primary identifier. 

### Changes Included
* **Database Migrations:** 
  * Added `domain` column to the `panel` table (VARCHAR, 20).
  * Safely backfilled all existing panels to the `Clinical` domain (Dependency 1).
  * Applied `NOT NULL` constraints post-backfill.
  * (Explicitly avoided bringing over the rejected `panel_lab_unit` foreign keys).
* **Hibernate & Entities:** 
  * Updated `Panel.hbm.xml` mapping to include `domain`.
  * Updated `Panel.java` valueholder.
* **Test Fixtures:** 
  * Updated `panel.xml` DBUnit seed data with `domain="Clinical"` to satisfy new constraints during test execution.

## Related Issues
* **Replaces / Closes:** PR #2464 (The previous implementation branch had significant rebase debt and was built against the now-rejected v1 spec. This fresh branch supersedes it).
* **Epic:** https://uwdigi.atlassian.net/browse/OGC-224

## Next Steps (M2)
* Once this schema foundation lands, M2 will introduce the backend synchronization for `SAMPLETYPE_PANEL` derivation and the REST controllers scoped for the Test Catalog Management shell.

## Verification
- [x] Migrations run cleanly on an empty database.
- [x] Migrations run safely over existing data (backfill handles `NULL` states).
- [x] `PanelServiceTest` integration suite compiles and passes 100%.

## Related Issue

https://uwdigi.atlassian.net/browse/OGC-224

### CODE REVIEW

**Review by mozzy11 — COMMENTED** [review]
Thanks Derrick — the core of this PR is exactly right. The three-step migration (add nullable column → backfill existing panels to Clinical → then enforce NOT NULL) is the safe, professional way to add a required column, and updating every panel fixture in the same PR shows good discipline.

A few things need attention before this can go in:

- **Rebase onto develop.** PR #3859 (Test Catalog Completion v2) just merged and rewrote base.xml, BaseWebContextSensitiveTest and PatientDAORedirectIntegrationTest, which is why this branch shows conflicts. The person_pk id=2 CI failure may even disappear after the rebase, since #3859 touched that exact test.
- **Renumber the changeset.** 006-panel-domain.xml collides with the existing 006 file and develop's 3.5.x.x series is now at 058 — please rename to the next free number (059+) and move the include to the end of base.xml.
- **Drop the is_merged="false" fixture edits** if they still seem needed after rebasing. That column has a DB default of false, so DBUnit fixtures don't need it — and if a test still fails with a duplicate-key error, the right fix is the resyncSequence(...) helper on BaseWebContextSensitiveTest, not fixture edits. Keeping this PR strictly about panel.domain will make it much easier to review.

Once CI is green on a rebased branch this should be a quick approve — the schema work itself is solid.

**Comment by Agaba-derrick** [comment]
@mozzy11   CI is green, Reverted the un-necessary  changes ! , kindly take another look

File: src/main/resources/liquibase/3.5.x.x/059-panel-domain.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+    xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.8.xsd">
+
+    <changeSet id="059-add-domain-to-panel" author="Agaba-derrick">
```
Comment (mozzy11) [inline src/main/resources/liquibase/3.5.x.x/059-panel-domain.xml]: Numbering collision: develop already has `059-results-r1-unified-worklist.xml`, and the series currently ends at `063-collapse-variant-groups.xml`. Rename this file (and the changeset id prefixes) to `064-panel-domain.xml`.

File: src/main/resources/liquibase/3.5.x.x/059-panel-domain.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        <comment>Add required domain column to panel table (OGC-1142 v2.2)</comment>
+        <addColumn tableName="panel" schemaName="clinlims">
```
Comment (mozzy11) [inline src/main/resources/liquibase/3.5.x.x/059-panel-domain.xml:10]: Missing precondition. Sibling changesets (e.g. `058-test-variant-link.xml`) guard DDL with `<preConditions onFail="MARK_RAN"><not><columnExists .../></not></preConditions>` so re-runs and distro DBs that already carry the column don't fail. Add the same guard here and on the NOT NULL changeset.

File: src/main/resources/liquibase/3.5.x.x/059-panel-domain.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            <column name="domain" type="varchar(20)">
+                <constraints nullable="true"/>
+            </column>
+        </addColumn>
+    </changeSet>
+
+    <changeSet id="059-backfill-panel-domain-clinical" author="Agaba-derrick">
+        <comment>Backfill existing panels to Clinical domain per v2.2 FRS</comment>
+        <update tableName="panel" schemaName="clinlims">
```
Comment (mozzy11) [inline src/main/resources/liquibase/3.5.x.x/059-panel-domain.xml:19]: Two things on the backfill: (1) it writes `'Clinical'` (title case) — the established pattern (`test.domain` in 040-test-domain-amr-whonet.xml, `test_section.domain` in 059-results-r1) is uppercase `'CLINICAL'` with a `CHECK (domain IN ('CLINICAL','ENVIRONMENTAL','VECTOR'))` and a column default. Please match it so cross-table domain filters don't need casing special-cases. (2) `<update>` has no auto-rollback — add an explicit `<rollback/>` so tag rollbacks across this file don't abort.

File: src/main/resources/liquibase/3.5.x.x/059-panel-domain.xml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            <column name="domain" value="Clinical"/>
+            <where>domain IS NULL</where>
+        </update>
+    </changeSet>
+
+    <changeSet id="059-domain-not-null" author="Agaba-derrick">
```
Comment (mozzy11) [inline src/main/resources/liquibase/3.5.x.x/059-panel-domain.xml]: Applying NOT NULL while no production code sets `domain` breaks panel creation at runtime: `PanelCreateController.createPanel`, `PanelCreateRestController.createPanel`, `TestCatalogEditorRestController` (~line 1762) and `OclToOpenElisMapper.createPanel` all insert panels without a domain. Either set the default in those write paths in this PR, add `defaultValue` on the column and relax the hbm mapping, or defer the NOT NULL to M2 when the write paths are updated.

File: src/main/resources/hibernate/hbm/Panel.hbm.xml [hunk 5]
```diff
@@ -24,6 +24,9 @@
         <property name="loinc" type="java.lang.String">
             <column name="LOINC" length="10" />
         </property>
+        <property name="domain" type="java.lang.String">
```
Comment (mozzy11) [inline src/main/resources/hibernate/hbm/Panel.hbm.xml:27]: `not-null="true"` makes Hibernate reject any Panel insert with unset `domain` before SQL is even issued — this is what turns the schema change into a runtime regression for the Add Panel flows. If NOT NULL stays in M1, every `new Panel()` call site in src/main/java must set a domain.

**Review by mozzy11 — CHANGES_REQUESTED** [review]
Thanks @Agaba-derrick — the three-changeset structure (add column → backfill → NOT NULL) is the right shape for OGC-1142 M1, the hbm mapping matches the constraint, and the fixture updates are thorough.

Four things block this, though — the first two are runtime/data regressions:
1. **Every production panel-creation path breaks**: `PanelCreateController`, `PanelCreateRestController`, `TestCatalogEditorRestController`, and `OclToOpenElisMapper` all insert a `Panel` without `domain`; with NOT NULL + `not-null="true"` those inserts throw at runtime. Only test code was updated.
2. **Domain value format diverges from the house pattern**: `test.domain` (OGC-936, changeset 040) and `test_section.domain` (059-results-r1) use **uppercase `CLINICAL`** with a `CHECK (domain IN ('CLINICAL','ENVIRONMENTAL','VECTOR'))` and a column default — this PR backfills title-case `'Clinical'` with no CHECK and no default. Cross-table domain filtering shouldn't need panel-specific casing.
3. Filename/prefix collision with develop's existing `059-*`; series is at 063 → renumber to 064.
4. Missing `preConditions onFail="MARK_RAN"` guards and a rollback for the `<update>`.

Details inline.

**Comment by Agaba-derrick** [comment]
@mozzy11 Are we able to kindly take another look ?

File: src/main/resources/liquibase/3.5.x.x/064-panel-domain.xml [hunk 6]
```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+    xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.8.xsd">
+
+    <changeSet id="064-add-domain-to-panel" author="Agaba-derrick">
```
Comment (mozzy11) [inline src/main/resources/liquibase/3.5.x.x/064-panel-domain.xml]: Develop's series moved again: #3625 merged `064`–`067` (starting with `064-add-whonet-code-to-type-of-sample.xml`), so this collides. Please rename the file and the three changeset id prefixes to `068-*` when rebasing.

File: src/main/resources/liquibase/3.5.x.x/base.xml [hunk 7]
```diff
@@ -168,4 +168,7 @@
   <include relativeToChangelogFile="true" file="062-result-limits-sample-type-scope.xml"/>
   <include relativeToChangelogFile="true" file="063-collapse-variant-groups.xml"/>
 
+  <!-- OGC-1142 v2.2: Add domain to panel table -->
+  <include relativeToChangelogFile="true" file="064-panel-domain.xml"/>
```
Comment (mozzy11) [inline src/main/resources/liquibase/3.5.x.x/base.xml]: This include is where the merge conflict lands — develop's `base.xml` now ends at `067-sample-type-disposal-instructions.xml`. After rebasing, place the renumbered `068-panel-domain.xml` include after 067.

File: src/test/resources/testdata/test-modify-service.xml [hunk 8]
```diff
@@ -104,9 +104,11 @@
     <sampletype_test id="1" sample_type_id="100"
         test_id="1" is_panel="false" />
 
+    <!-- domain="Clinical" required: panel.domain is NOT NULL since 059-panel-domain
```
Comment (mozzy11) [inline src/test/resources/testdata/test-modify-service.xml]: Stale comment: the migration is no longer `059-panel-domain`, and the value is uppercase `CLINICAL`, not `Clinical`. Please update it to match the renumbered changeset.

**Review by mozzy11 — CHANGES_REQUESTED** [review]
Thanks @Agaba-derrick — all four blocking items from the last review are addressed: every production panel write path now sets `domain` (plus a `CLINICAL` default in the `Panel()` constructor), the backfill uses uppercase `CLINICAL` with the house `CHECK` constraint and column default, all three changesets carry `preConditions onFail="MARK_RAN"` guards and rollbacks, and the file was renumbered off 059.

What still blocks this is timing: develop moved again. #3625 (Sample Type Management) merged and took 064–067 (`064-add-whonet-code-to-type-of-sample.xml` … `067-sample-type-disposal-instructions.xml`), so `064-panel-domain.xml` collides once more and the branch is CONFLICTING against develop — it must be rebased. Please renumber to 068, move the `base.xml` include after 067, and rebase; the E2E gate also failed on the last run (Playwright core/harness), so we need a green run on the rebased branch. One stale fixture comment noted inline. The schema work itself is in good shape — this should be quick to land once renumbered and green.

**Comment by mherman22** [comment]
Sweep note: I can't re-review this yet, @Agaba-derrick — two things block it independently of the review feedback:

1. The branch is **CONFLICTING** with `develop` (bot flagged it again on 2026-07-28). Please rebase.
2. CI is **failing** on the current head, so even after the rebase there's no green build to approve against.

Once both are cleared, re-ping and this goes back in the queue — the outstanding `CHANGES_REQUESTED` review also needs its points addressed in the same push.

**Comment by Agaba-derrick** [comment]
@mherman22 do you wanna checkout this ?

**Comment by HerbertYiga** [comment]
@Agaba-derrick fix up the conflicts on this!

**Comment by Agaba-derrick** [comment]
This PR is blocked from a green CI run by a pre-existing upstream bug:
- **Blocks:** #4042 — `ResultEntryRestControllerTest` failures (NoteService duplicate + FhirContext NPE in test env)

The two failing tests are **not caused by this PR**. They were introduced in `develop` by the Results Entry v3 merge and are tracked separately.

**Comment by Agaba-derrick** [comment]
@mozzy11  does #4039 solve https://uwdigi.atlassian.net/browse/OGC-224 we can close this one if everything is solved
